endpoint/endpoint.go

```diff
@@ -26,6 +26,14 @@ const (
 	OwnerLabelKey = "owner"
 )
 
+// TTL is a structure defining the TTL of a DNS record
+type TTL struct {
+	// The value of the TTL
+	Value int64
+	// Whether or not the TTL value is configured
+	IsConfigured bool
```

Comment:
I was following the principle of being explicit rather than implicit. This is to distinguish between a TTL being configured (in the annotation) or not. If not, it should take the default of the zone. I don't want to denote a special value (like `-1`) or make ttl `*int64` and use `nil` to mean TTL not configured.

From your comment below, it looks like you prefer having `ttl=0` to mean it's not configured?